### PR TITLE
[url_launcher] Adds macOS package with example app

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

Adds the macos implementation of the url_launcher plugin.
Tested by adding an example app that runs the e2e tests. This example app is a copy of the url_launcher/example app, with a modified pubspec.yaml that depends on url_launcher from pub, and url_launcher_macos from path. There is also a minor change in main.dart where the last widget is removed since there is no macOS implementation for that method.

The Swift implementation of the plugin was ported from https://github.com/google/flutter-desktop-embedding/tree/master/plugins/flutter_plugins/url_launcher_fde

## Related Issues

flutter/flutter#41721

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/url_launcher/url_launcher_macos/macos/Classes/UrlLauncherPlugin.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (amirh) [inline packages/url_launcher/url_launcher_macos/macos/Classes/UrlLauncherPlugin.swift]: Should this be the the standard license header we use?

Comment (franciscojma86) [reply packages/url_launcher/url_launcher_macos/macos/Classes/UrlLauncherPlugin.swift]: Yeah, updated

**Comment by domesticmouse** [comment]
Any idea when this will be published on pub.dev? I have a [desktop sample app](https://github.com/flutter/samples/pull/174) in progress that uses `url_launcher` and I'd love to drop the reference to `url_launcher_fde`.

**Comment by franciscojma86** [comment]
@domesticmouse It is on pub.dev now, but not endorsed yet. You can still use it though.

**Comment by domesticmouse** [comment]
@franciscojma86 I have no idea what endorsed means, but using `url_launcher: ^5.3.0` without `url_launcher_fde` results in a macOS desktop binary that throws `MissingPluginException (MissingPluginException(No implementation found for method canLaunch on channel plugins.flutter.io/url_launcher))`

**Comment by franciscojma86** [comment]
Endorsed means that you would just be able to use `url_launcher: ^5.3.0` and should work, but in this case you have to do a bit more. Try adding to your pubspec:

```
url_launcher: ^5.3.0
url_launcher_macos: ^0.0.1
```

**Pull request merged**